### ISSUE TITLE
Exploring Zero Noise Extrapolation (ZNE) for circuit cutting

### ISSUE THREAD

**Issue by ritajitmajumdar1** [qiskit-advocate/qamp-spring-23#13]
### Description

Circuit cutting [1] has been proposed as a method to partition a circuit into multiple smaller sub-circuits or fragments. Each fragment can be executed independent of the other, and the final probability distribution of the uncut circuit is computed via classical postprocessing over the individual probability distributions of the smaller fragments. However, the classical postprocessing time scales exponentially in the number of qubits. So, circuit cutting is a feasible approach only when a circuit can be efficiently partitioned using a small number of cuts.

In general, each sub-circuit is expected to have lower noise since they have fewer qubits and gates. Therefore, circuit cutting alone is expected to be a method to lower noise in the system [2,3]. In [4], we studied error mitigation on tomographic approach of circuit cutting. The mitigation technique used there is specific to tomography (or similar methods which provides a description of the channel) and is not applicable to general circuits. A study of error mitigation on generic circuit cutting (not its tomographic variant) is largely missing. In this project we want to study the effect of Zero Noise Extrapolation (ZNE) on circuit cutting.

ZNE is a biased error mitigation technique which computes the expectation value of an observable at different noise scales and extrapolates it to determine the expectation value of the observable at zero noise limit. In this study we shall apply ZNE on each fragment obtained after circuit cutting, and find the expectation value of the full circuit at zero noise limit. The aim is to understand the effectiveness and the overhead of applying ZNE on circuit cutting; whether ZNE on each sub-circuit can indeed produce expectation values closer to the ideal one than ZNE on the entire circuit.

[1] Peng, Tianyi, et al. "Simulating large quantum circuits on a small quantum computer." Physical review letters 125.15 (2020): 150504.
[2] Ayral, Thomas, et al. "Quantum divide and compute: exploring the effect of different noise sources." SN Computer Science2.3 (2021): 132.
[3] Basu, Saikat, et al. "i-qer: An intelligent approach towards quantum error reduction." ACM Transactions on Quantum Computing 3.4 (2022): 1-18.
[4] Majumdar, Ritajit, and Christopher J. Wood. "Error mitigated quantum circuit cutting." arXiv preprint arXiv:2211.13431(2022).

### Deliverables

- Python code implementing numerical experiments
- Numerical results

### Mentors details

- Mentor 1
  - Name: Ritajit Majumdar
  - GitHub ID: @ritajitmajumdar1
  - What they do: Research Scientist
- Mentor 2
  - Name: Derek Wang
  - GitHub ID: @derek-wang-ibm
  - What they do: Research Scientist

### Number of mentees

2

### Type of mentees

**Disclaimer: The final number of mentees required for this project is yet to be determined. We may be able to accommodate upto 2 mentees. In case we consider a second mentee, the requirement will remain the same as that of the first mentee.**

- Mentee 1
  - Required:
    - Working knowledge of quantum information theory, as covered in chapters 2 and 8 of the textbook by Nielsen and Chuang
    - Proficient experience of coding in python and qiskit
    - Knowledge of error mitigation, especially zero noise extrapolation
    - Experience of performing independent and original research
    - Willing to invest time beyond the QAMP project period, if necessary, to produce a publishable version (or a white paper, depending on the outcome) of the research
  - Nice to have:
    - Have read the following paper which gives a nice introduction to quantum circuit cutting: [https://arxiv.org/abs/2012.02333]
    - Propose their own ideas on how to implement ZNE with circuit cutting, potential challenges and probable pitfalls

**Comment by ssawarn** [comment]
I would like to work on the project. I have a background in physics and is proficient in qiskit. Also I am ready to invest more time. I have dm you on slack with my cv @ritajitmajumdar1 .

**Comment by hsanthan** [comment]
@ritajitmajumdar1 , I am interested in this project. Quantum information theory by Nielsen and Chuang is one of the first books through which I got introduced to Quantum Computing and I'm more than happy to take a deeper dive and learn more. I am fairly good at qiskit modules and have earned the qiskit developer certification. 

I have been a mentee in QAMP22 and extended the project work into an MDPI journal manuscript that is soon going to be published. I'm willing to work beyond the QAMP timelines/deliverables and invest extra time. Thanks!

**Comment by ritajitmajumdar1** [comment]
Hi @hsanthan, did you apply for the project? Please apply for the project via the airtable link shared to the qiskit advocates.

**Comment by hsanthan** [comment]
Yes @ritajitmajumdar1, I applied for the project just a little while back.

**Comment by ritajitmajumdar1** [comment]
@hsanthan I don't see your name or any contact details in the Airtable. Can you provide your email here?

**Comment by hsanthan** [comment]
@ritajitmajumdar1, here are my details:
Name: Hemavathi Santhanam
Email: hsanthan@ibm.com  

I've submitted the form again now. Please let me know if you find my details.

**Comment by GemmaDawson** [comment]
@gideonuchehara - please add a comment to this issue so that I may assign it to you

**Comment by gideonuchehara** [comment]
Thanks @GemmaDawson

**Comment by gideonuchehara** [comment]
[Exploring Zero Noise Extrapolation (ZNE) for circuit cutting.pptx](https://github.com/qiskit-advocate/qamp-spring-23/files/11488939/Exploring.Zero.Noise.Extrapolation.ZNE.for.circuit.cutting.pptx)

**Comment by gideonuchehara** [comment]
[Exploring_Zero_Noise_Extrapolation__ZNE__for_Circuit_Cutting.pdf](https://github.com/qiskit-advocate/qamp-spring-23/files/12008493/Exploring_Zero_Noise_Extrapolation__ZNE__for_Circuit_Cutting.pdf)

**Comment by gideonuchehara** [comment]
[ZNE_for_Circuit_Cutting_final_presentation.pptx](https://github.com/qiskit-advocate/qamp-spring-23/files/12105516/ZNE_for_Circuit_Cutting_final_presentation.pptx)
